### PR TITLE
Fix: Remove primary key from page_stat

### PR DESCRIPTION
**Pull request by robert00s**
Fix incorrect migration from v1 to v2.
Table page_stat only allowed one record per book.

### CODE REVIEW

**Comment by poire-z** [comment]
You may wish to add an index on the keys you are going to query on often.
`CREATE INDEX IF NOT EXISTS page_stat_id_book ON page_stat(id_book);`
to replace the implicit index created for your primary key.
It will speed up query by book_id when there are a lot of lines in this table (it will not have to do a full table scan).

May be too on your book table
`CREATE INDEX IF NOT EXISTS book_title_authors_md5 ON book(title, authors, md5);`
if you use these 3 columns in most of ur queries.

https://sqlite.org/lang_createindex.html
(Did that in https://github.com/robert00s/koreader/blob/f01b81b52bdadb36bd4e065f138bfee93ef30fa7/plugins/coverbrowser.koplugin/bookinfomanager.lua#L61 )

**Comment by robert00s** [comment]
@poire-z Thanks. It's really great tip.

**Comment by Frenzie** [comment]
Couldn't you just stick that in the `CREATE TABLE` statement itself?

```sql
         CREATE TABLE IF NOT EXISTS page_stat
             (
--blabla
                 FOREIGN KEY(id_book) REFERENCES book(id),
                 INDEX page_stat_id_book (id_book)
             );
```

**Comment by poire-z** [comment]
Seems not possible with sqlite: Error: near "INDEX": syntax error
https://sqlite.org/lang_createtable.html does not mention the option of creating an index in the create table statement.

**Comment by Frenzie** [comment]
Fair enough. I'm reasonably sure it's possible in MySQL and PostgreSQL.

File: plugins/statistics.koplugin/main.lua [hunk 1]
```diff
@@ -929,7 +937,7 @@ function ReaderStatistics:getReadingProgressStats(sdays)
         ORDER  BY dates DESC
     ]]
     local result_book = conn:exec(string.format(sql_stmt, period_begin))
-
+    if result_book == nil then return nil end
```
Comment (Frenzie) [inline plugins/statistics.koplugin/main.lua]: If you just return then you always return nil. :-) Btw, is it important to check for nil or could it be simplified to `if not result_book then return end`?

**Pull request merged**